### PR TITLE
feat: Allow early logger initialization

### PR DESCRIPTION
**Pull request by guybedford**
This resolves https://github.com/fastly/js-compute-runtime/issues/225 in supporting lazy initialization of the logging endpoint allowing the logger to be created during preinitialization.

### CODE REVIEW

**Comment by LeslieCarr** [comment]
@jaskiratr

**Review by elliottt — COMMENTED** [review]
I think this all seems reasonable, but I wonder if the additional logging to stderr should be configurable.

File: runtime/fastly/builtins/logger.cpp [hunk 1]
```diff
@@ -9,13 +9,40 @@ namespace fastly::logger {
...
+  host_api::LogEndpoint endpoint(endpoint_id.toInt32());
 
   auto msg = core::encode(cx, args.get(0));
   if (!msg) {
     return false;
   }
 
+  fprintf(stdout, "Log [%s]: %s\n", endpoint_name_str.ptr.get(), msg.ptr.get());
+  fflush(stdout);
+
```
[the remark this answers is not in the document]
Comment (guybedford) [reply runtime/fastly/builtins/logger.cpp:45]: Apparently this behaviour is to match the Rust SDK, and that the behaviour is desireable by default for JS and Go.

Having a way to turn it off might well make sense, and agreed `enableDebugLogging()` sounds like a good integration point. I do not know about the design of `enableDebugLogging()` though.

[the remark this answers is not in the document]
Comment (guybedford) [reply runtime/fastly/builtins/logger.cpp:45]: Actually `enableDebugLogging` is off by default so doesn't fit the requirements here unfortunately?

**Review by elliottt — APPROVED** [review]
As this is matching existing rust SDK behavior, it all seems good to me!

**Comment by guybedford** [comment]
Looking into this further, it seems Viceroy does handle the logging to stdout as I would have expected, so this PR actually causes a double log when `--verbose` is set.

Instead I'm just going to refactor this to not do the stdout logging and just be a fix for #225.

**Comment by guybedford** [comment]
I've updated this PR title and description to just be a fix for #225.

File: integration-tests/js-compute/fixtures/app/src/out-of-memory.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    if (starlingmonkey) {
+        return pass()
+    }
+    try {
+        const objects = [];
+        for (let i = 0; i < 1000; i++) {
+            objects.push(new BigUint64Array(1_000_000));
+        }
+    }
+    return fail('expected out of memory error')
```
Comment (JakeChampion) [inline integration-tests/js-compute/fixtures/app/src/out-of-memory.js:17]: Does `fail` need to be imported?

Comment (guybedford) [reply integration-tests/js-compute/fixtures/app/src/out-of-memory.js:17]: Thanks for catching this, this wasn't supposed to be committed, and comes from https://github.com/fastly/js-compute-runtime/pull/805, but wasn't landed as we don't currently have a good stdout testing mechanism.

**Review by elliottt — APPROVED** [review]

**Pull request merged**